src/squidpy/gr/_build.py

```diff
@@ -349,3 +464,116 @@ def _transform_a_spectral(a: spmatrix) -> spmatrix:
 
 def _transform_a_cosine(a: spmatrix) -> spmatrix:
     return cosine_similarity(a, dense_output=False)
+
+
+@d.dedent
+def mask_graph(
```

Comment:
tested it and seems to work, but how could this be used downstream for `var_by_distance` tasks? If you mask your neighbor graph and then calculate distances from some anchor point, wouldn't it again "draw" the lines through the area that was previously masked? How to make sure that only the positive mask is used for later distance calculations?

Reply:
As mentioned during the call, one could traverse a sufficiently dense enough neighborhood graph. We'd have to evaluate if this works though.

Reply:
> tested it and seems to work, but how could this be used downstream for var_by_distance tasks? 

as it's currently implemented `var_by_distance`, it can't . See https://github.com/scverse/squidpy/issues/837 . Var by distance rely on computing the 1-nn point to the closest annotation, and shapely can handle that with `distance`https://shapely.readthedocs.io/en/stable/reference/shapely.distance.html#shapely.distance .

> How to make sure that only the positive mask is used for later distance calculations?

for `var_by_distance` it wouldn't, in fact `var_by_distance` doesn't uses the graph stored in `obsp` but recomputed the 1-nn base on the annotations. How to handle this behaviour in `var_by_distance` is more complicated cause then 2 vector geometries should be passed. One for the region to compute the distance from, and one to define which regions should not be crossed (the empty space). Nevertheless, this behaviour is useful for stuff like e.g. nhood_enrichment or other methods that rely on the graph calculated here.